File: backend/app/models/project.py
import enum

from sqlalchemy import Column, Integer, String, Enum, ForeignKey, Index
from sqlalchemy.orm import relationship

from app.database import Base
from app.models.base import TimestampMixin


class Region(str, enum.Enum):
    TN = "TN"
    Kerala = "Kerala"
    AP = "AP"
    Telangana = "Telangana"
    Gujarat = "Gujarat"
    Delhi = "Delhi"
    Mumbai = "Mumbai"


class Category(str, enum.Enum):
    FMCG = "FMCG"
    Industrial_Goods = "Industrial Goods"


class ProjectStatus(str, enum.Enum):
    brand_description_generated = "Brand description generated"
    deck_in_progress = "Deck in progress"
    deck_shared = "Deck Shared"
    client_approved = "Client approved"
    client_rejected = "Client rejected"
    video_production_in_progress = "Video production in progress"
    video_submitted_for_review = "Video submitted for review"
    video_approved = "Video approved"
    campaign_signed_up = "Campaign signed up"


class Project(Base, TimestampMixin):
    __tablename__ = "projects"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    region = Column(Enum(Region), nullable=False)
    city = Column(String(100), nullable=False)
    salesperson_name = Column(String(150), nullable=False)
    brand_name = Column(String(200), nullable=False)
    category = Column(Enum(Category), nullable=False)
    status = Column(
        Enum(ProjectStatus),
        default=ProjectStatus.brand_description_generated,
        nullable=False,
    )

    created_by_user = relationship("User", back_populates="projects")

    __table_args__ = (
        Index("ix_projects_region", "region"),
        Index("ix_projects_status", "status"),
        Index("ix_projects_category", "category"),
        Index("ix_projects_user_id", "user_id"),
    )

File: frontend/src/types/project.ts
export type Region = 'TN' | 'Kerala' | 'AP' | 'Telangana' | 'Gujarat' | 'Delhi' | 'Mumbai';

export type Category = 'FMCG' | 'Industrial Goods';

export type ProjectStatus =
  | 'Brand description generated'
  | 'Deck in progress'
  | 'Deck Shared'
  | 'Client approved'
  | 'Client rejected'
  | 'Video production in progress'
  | 'Video submitted for review'
  | 'Video approved'
  | 'Campaign signed up';

export interface Project {
  id: number;
  user_id: number;
  region: Region;
  city: string;
  salesperson_name: string;
  brand_name: string;
  category: Category;
  status: ProjectStatus;
  created_at: string;
  updated_at: string | null;
}

export interface ProjectCreateData {
  region: Region;
  city: string;
  salesperson_name: string;
  brand_name: string;
  category: Category;
  status?: ProjectStatus;
}

export interface ProjectUpdateData {
  region?: Region;
  city?: string;
  salesperson_name?: string;
  brand_name?: string;
  category?: Category;
  status?: ProjectStatus;
}

export interface ProjectListResponse {
  items: Project[];
  total: number;
  page: number;
  per_page: number;
}

export interface RegionCount {
  region: string;
  count: number;
}

export interface DashboardMetrics {
  total_projects: number;
  clients_by_region: RegionCount[];
  briefs_approved: number;
  videos_generated: number;
  videos_approved: number;
  campaigns_completed: number;
}

export const REGIONS: Region[] = ['TN', 'Kerala', 'AP', 'Telangana', 'Gujarat', 'Delhi', 'Mumbai'];

export const CATEGORIES: Category[] = ['FMCG', 'Industrial Goods'];

export const PROJECT_STATUSES: ProjectStatus[] = [
  'Brand description generated',
  'Deck in progress',
  'Deck Shared',
  'Client approved',
  'Client rejected',
  'Video production in progress',
  'Video submitted for review',
  'Video approved',
  'Campaign signed up',
];
